plugins/optimization-detective/types.ts

```diff
@@ -69,11 +82,16 @@ export type FinalizeArgs = {
 		properties: ExtendedElementData
 	) => void;
 	readonly isDebug: boolean;
+	readonly log: LogFunction;
+	readonly info: LogFunction;
+	readonly warn: LogFunction;
+	readonly error: LogFunction;
 };
 
 export type FinalizeCallback = ( args: FinalizeArgs ) => Promise< void >;
 
 export interface Extension {
+	readonly name?: string;
 	initialize?: InitializeCallback;
 	finalize?: FinalizeCallback;
```

Comment:
Oh, these should have been `readonly` too, right?

```suggestion
	readonly initialize?: InitializeCallback;
	readonly finalize?: FinalizeCallback;
```

Reply:
Yes, that makes sense.